pkg/flow/internal/controller/component.go

```diff
@@ -180,21 +191,23 @@ func getManagedOptions(globals ComponentGlobals, cn *ComponentNode) component.Op
 
 	wrapped := newWrappedRegisterer()
 	cn.register = wrapped
+
 	return component.Options{
 		ID:     globalID,
-		Logger: logging.New(logging.LoggerSink(globals.Logger), logging.WithComponentID(cn.nodeID)),
+		Logger: log.With(globals.Logger, "component", globalID),
 		Registerer: prometheus.WrapRegistererWith(prometheus.Labels{
 			"component_id": globalID,
 		}, wrapped),
-		Tracer:    wrapTracer(globals.TraceProvider, globalID),
+		Tracer:    tracing.WrapTracer(globals.TraceProvider, globalID),
 		Clusterer: globals.Clusterer,
 
 		DataPath:       filepath.Join(globals.DataPath, cn.nodeID),
 		HTTPListenAddr: globals.HTTPListenAddr,
 		DialFunc:       globals.DialFunc,
 		HTTPPath:       path.Join(prefix, cn.nodeID) + "/",
 
-		OnStateChange: cn.setExports,
+		OnStateChange:    cn.setExports,
+		ModuleController: globals.ModuleController,
```

Comment:
I flipped this around and now throw an error if id is not unique. This could lead to a situation where modules don't know what ID to pass. The current implementation forces modules to use `options.ID` which is unique when used appropriately. I think at some point when introduce multi loading then we can backtrack a bit. Though in general I think module loaders should always have the calling component ID in them so you can backtrack if given a log entry.